ideascube/utils.py

```diff
@@ -1,8 +1,27 @@
+from io import TextIOWrapper
 import sys
 
 from django.conf import locale
 
 
+class AutomagicTextIOWrapper(TextIOWrapper):
```

Comment:
>> This is why I suggest to just put it on the class itself: curious people will end there if they want to understand why we use it :)

> We use it for the exact same reasons we'd use a TextIOWrapper. Also, it's not a very specific class, and (if moved to use cchardet) could even be sent to upstream cchardet.

Coming back on this, because I just realized we might be approaching this from a different angle.

When I say « We use it for the exact same reasons we'd use a `TextIOWrapper` », I'm looking at it from the perspective that using a `TextIOWrapper`-like solution is the obvious thing to do.

It seems to me that what you're implying is that, for you, the obvious thing to do would have been to `splitlines` and use an array, which makes using a `TextIOWrapper` here a sort of special case in your view. Am I right?

If that is indeed your view, then I'm going to have to respectfully disagree with it, and here is why.

The right way to read files line by line (and we all know that TOOWTDI :stuck_out_tongue_winking_eye:) is:

```python
with open('path', 'r') as f:
    for line in f:
        ...
```

I would consider the following to be a mistake because it reads the whole file (which might be very big) at once and loads it all in memory, iterates over the content once to split it, then iterate again over the list of lines:

```python
with open('path', 'r') as f:
    lines = f.read().splitlines()

    for line in lines:
        ...
```

To me the above example naturally leads to using a generator-like solution to decoding the lines. `TextIOWrapper` is such a solution, and it is builtin with Python. Does that not make it the right choice?

All this to say that I find nothing surprising in using a `TextIOWrapper`-like solution here, and really don't see the need to specifically comment on its usage.

I would have instead found a solution based on `.read().splitlines()` more surprising and deserving of a comment. (« we can't just iterate over the file as we don't know its encoding » or something)

In fact, if you remember, [I did find it surprising and asked for details](https://github.com/ideascube/ideascube/pull/610#discussion_r85533166). :smile: